Nice to have more http headers

sakkiii opened the pull request:
## Description


## Motivation and Context
Added more security headers for better security
- `X-Frame-Options` tells the browser whether you want to allow your site to be framed or not. By preventing a browser from framing your site you can defend against attacks like clickjacking.
- `X-Content-Type-Options` stops a browser from trying to MIME-sniff the content type and forces it to stick with the declared content-type. The only valid value for this header is `"X-Content-Type-Options: nosniff"`.
- `HTTP Strict Transport Security` is an excellent feature to support on your site and strengthens your implementation of TLS by getting the User Agent to enforce the use of HTTPS.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Security Enhancements  

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

sakkiii committed:
Nice to have more http headers

Added more security headers for better security

sakkiii committed:
Merge branch 'master' into patch-1

sakkiii committed:
Merge branch 'master' into patch-1

aead reviewed (approved):
LGTM  - HSTS is acceptable since on non-TLS connections it's ignored by clients.

klauspost reviewed (approved):
LGTM

sakkiii committed:
Merge branch 'master' into patch-1

sakkiii committed:
Merge branch 'minio:master' into patch-1

sakkiii committed:
comment formated 

lint check failed `cmd/generic-handlers.go:567: File is not `gofmt`-ed with `-s` (gofmt)`

sakkiii committed:
Merge branch 'minio:master' into patch-1

sakkiii committed:
Update generic-handlers.go

sakkiii committed:
lint check fixed updated format

harshavardhana commented on cmd/generic-handlers.go:
```diff
@@ -564,11 +564,16 @@ func addCustomHeaders(h http.Handler) http.Handler {
+		header.Set("X-XSS-Protection", "1; mode=block")                                // Prevents against XSS attacks
+		header.Set("Content-Security-Policy", "block-all-mixed-content")               // prevent mixed (HTTP / HTTPS content)
+		header.Set("X-Frame-Options", "deny")                                          // Prevent against Clickjack attack
+		header.Set("X-Content-Type-Options", "nosniff")                                //  Prevent mime-sniff
```
```suggestion
		header.Set("X-Content-Type-Options", "nosniff")                                // Prevent mime-sniff
```

sakkiii replied:
Done thanks 👍

harshavardhana commented on cmd/generic-handlers.go:
```diff
@@ -564,11 +564,16 @@ func addCustomHeaders(h http.Handler) http.Handler {
+		header.Set("Content-Security-Policy", "block-all-mixed-content")               // prevent mixed (HTTP / HTTPS content)
+		header.Set("X-Frame-Options", "deny")                                          // Prevent against Clickjack attack
+		header.Set("X-Content-Type-Options", "nosniff")                                //  Prevent mime-sniff
+		header.Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains") // HSTS mitigates variants of MiTM attacks
```
```suggestion
		header.Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains") // HSTS mitigates variants of MITM attacks
```

sakkiii replied:
Done thanks 👍

sakkiii committed:
Update cmd/generic-handlers.go

Co-authored-by: Harshavardhana <harsha@minio.io>

sakkiii committed:
Update cmd/generic-handlers.go

comment format fix

Co-authored-by: Harshavardhana <harsha@minio.io>

sakkiii committed:
Merge branch 'master' into patch-1

harshavardhana committed:
Merge branch 'master' into patch-1
